libraries/c_sdk/standard/ble/test/iot_mqtt_ble_system_test.c

```diff
@@ -0,0 +1,1040 @@
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file core_mqtt_system_test.c
+ * @brief Integration tests for the coreMQTT library for the FreeRTOS platform
+ * over Secure Sockets as the transport layer.
+ */
+
+/* Standard header includes. */
+#include <string.h>
+#include <stdbool.h>
+#include <assert.h>
+#include <stdlib.h>
+#include <time.h>
+
+/* Unity Test framework include. */
+#include "unity_fixture.h"
+
+/* Include connection configurations header. */
+#include "aws_clientcredential.h"
+
+/* Include for Semaphore. */
+#include "platform/iot_threads.h"
+
+/* Include paths for public enums, structures, and macros. */
+#include "core_mqtt.h"
+#include "core_mqtt_state.h"
+
+#include "iot_ble.h"
+
+/* Include BLE implementation of transport interface. */
+#include "iot_ble_mqtt_transport.h"
+
+/* Include task.h for delay function. */
+#include "task.h"
+
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
+
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for DEMO.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for DEMO.
+ */
+
+#include "logging_levels.h"
+
+/* Logging configuration for the Demo. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "TEST"
+#endif
+
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
+#endif
+#include "logging_stack.h"
+
+/**************Default Configurations values***********************/
+
+/**
+ * This file contains 2 test groups:
+ * - coreMQTT_Integration_AWS_IoT_Compatible for running tests against AWS IoT
+ * - coreMQTT_Integration for running tests against a non-AWS IoT MQTT 3.1.1 MQTT broker.
+ *
+ * Thus, the 2 test groups have different broker endpoint and TLS credentials
+ * for running to connect against their brokers. However, this test shares the same
+ * macros (BROKER_ENDPOINT, BROKER_PORT, SERVER_ROOT_CA_CERT, and macros in
+ * tests/include/aws_clientcredential_key.h file) for the test groups.
+ * Thus, ONLY ONE test group (either against AWS IoT or a different broker) can be run
+ * for a single build depending on the credentials provided.
```

Comment:
These comment block about 2 different test groups may not be true for this test file.

Reply:
That comment is no longer relevant. Removed it.